### PR TITLE
[JSON] complete the dsl constraint

### PR DESCRIPTION
**Pull request by jswong65**
In this PR, mostly tried to complete the dsl constraint. Although, it would introduce trailing comma at the end of a constraint and a constraintset. It can still generate a valid JSON syntax. Validate all the new tests on [JSON Formatter & Validator](https://jsonformatter.curiousconcept.com/)

New tests: testAnchor01-03, testConstraint01-03, and testConstraintSet01-02

### CODE REVIEW

File: constraintlayout/core/src/main/java/androidx/constraintlayout/core/dsl/Constraint.java [hunk 1]
```diff
@@ -38,9 +43,39 @@ public class Anchor {
...
         Constraint getParent() {
             return Constraint.this;
         }
+
+        @Override
+        public String toString() {
+            StringBuilder ret = new StringBuilder("[");
+
+            if (mConnection != null) {
+                ret.append("'").append(mConnection.getId()).append("',")
```
Comment (jafu888) [inline constraintlayout/core/src/main/java/androidx/constraintlayout/core/dsl/Constraint.java:59]: For efficiency 
have a build(StringBuilder builder)  method.
that you use

File: constraintlayout/core/src/main/java/androidx/constraintlayout/core/dsl/Constraint.java [hunk 2]
```diff
@@ -98,15 +140,164 @@ enum Side {
...
+                }
+                ret.append("],\n");
+            }
+        }
+        if (editorAbsoluteX != Integer.MIN_VALUE) {
+            ret.append("editorAbsoluteX:").append(editorAbsoluteX).append(",\n");
+        }
+        if (editorAbsoluteY != Integer.MIN_VALUE) {
+            ret.append("editorAbsoluteY:").append(editorAbsoluteY).append(",\n");
+        }
```
Comment (jafu888) [inline constraintlayout/core/src/main/java/androidx/constraintlayout/core/dsl/Constraint.java:235]: remove editorAbsoluteX/Y makes no sense from a API.

File: constraintlayout/core/src/main/java/androidx/constraintlayout/core/dsl/Constraint.java (hunk 1 above)
Comment (jswong65) [reply constraintlayout/core/src/main/java/androidx/constraintlayout/core/dsl/Constraint.java:82]: Done!

File: constraintlayout/core/src/main/java/androidx/constraintlayout/core/dsl/Constraint.java [hunk 3]
```diff
@@ -98,15 +140,164 @@ enum Side {
...
+        if (start.mConnection != null) {
+            ret.append("start:").append(start).append(",\n");
+        }
+        if (end.mConnection != null) {
+            ret.append("end:").append(end).append(",\n");
+        }
+        if (baseline.mConnection != null) {
+            ret.append("baseline:").append(baseline).append(",\n");
+        }
```
[the remark this answers is not in the document]
Comment (jswong65) [reply constraintlayout/core/src/main/java/androidx/constraintlayout/core/dsl/Constraint.java:1033]: Done!

**Review by jswong65 — COMMENTED** [review]
Thanks for the suggestions! Addressed most of the comments. I will handle helper in the other PR. Will keep the variables untouched for now.

**Review by jafu888 — COMMENTED** [review]
Small nit then we submit

**Pull request merged**